service/application/src/test/java/com/solace/maas/ep/event/management/agent/service/ScanServiceTests.java

```diff
@@ -339,4 +341,84 @@ public void testFindById() {
 
         assertThatNoException();
     }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"COMPLETE", "FAILED", "TIMED_OUT"})
+    @SneakyThrows
+    public void testIsScanCompleteIsComplete(String scanStatus) {
+        ScanStatusEntity scanStatusA = scanServiceHelper.buildScanStatusEntity("status1", scanStatus);
+        ScanTypeEntity scanTypeA = scanServiceHelper.buildScanTypeEntity("123", "queueListing", null, scanStatusA);
+
+        when(scanTypeRepository.findAllByScanId(any(String.class)))
+                .thenReturn(List.of(scanTypeA));
+        when(scanStatusRepository.findByScanType(any(ScanTypeEntity.class)))
+                .thenReturn(scanStatusA);
+
+        Assertions.assertTrue(scanService.isScanComplete("scan1"));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"IN_PROGRESS", "INITIATED"})
+    @SneakyThrows
+    public void testIsScanCompleteIsNotComplete(String scanStatus) {
+        ScanStatusEntity scanStatusA = scanServiceHelper.buildScanStatusEntity("status1", scanStatus);
+        ScanTypeEntity scanTypeA = scanServiceHelper.buildScanTypeEntity("123", "queueListing", null, scanStatusA);
+
+        when(scanTypeRepository.findAllByScanId(any(String.class)))
+                .thenReturn(List.of(scanTypeA));
+        when(scanStatusRepository.findByScanType(any(ScanTypeEntity.class)))
+                .thenReturn(scanStatusA);
+
+        Assertions.assertFalse(scanService.isScanComplete("scan1"));
+    }
+
+    @Test
+    @SneakyThrows
+    public void testIsScanCompleteIsNotCompleteEmptyScanTypeList() {
```

Comment:
The names in this test class is a bit confusing, I think what we are trying to say is 
`{"COMPLETE", "FAILED", "TIMED_OUT"}` are `FINAL` statuses indicating the scan is complete 
and `{"IN_PROGRESS", "INITIATED"})` indicating otherwise. 
so we should adopt the names accordingly